src/java.management.rmi/share/classes/javax/management/remote/rmi/RMIConnectionImpl.java

```diff
@@ -1292,10 +1298,20 @@ public NotificationResult run() {
                         return getServerNotifFwd().fetchNotifs(csn, t, mn);
                     }
             };
-            if (acc == null)
-                return action.run();
-            else
-                return AccessController.doPrivileged(action, acc);
+            if (!SharedSecrets.getJavaLangAccess().allowSecurityManager()) {
+                // Modern case
+                if (subject == null)
+                    return action.run();
+                else
+                    return Subject.doAs(subject, action);
```

Comment:
Code style comment: put braces around the if/else to be consistent with lines below.